ICU-21480 update pom.xml to point to CLDR maven

srl295 opened the pull request:
- updated README as well
- deleted install-cldr-jars.sh

aheninger commented:
We forgot to copy this change over to the ICU4C copy of the test data file. Could you do that too?

pedberg-icu commented:
Thanks, I will copy over to the ICU4C file

cgohlke committed:
ICU-21210 Fix static builds on Windows

aheninger committed:
ICU-21346 Update user guide section on initialization.

This change is narrowly focused on the initialization and cleanup / termination
of ICU. The whole user-guide section (file) needs to be reviewed for out-of-date
information, but not as part of this change.

echeran committed:
ICU-21117 Use Bazel to automate generation of Unicode data files

jefgen committed:
ICU-21474 Update top-level README status badge to replace Travis CI with GitHub Actions.

pedberg-icu committed:
ICU-21480 integrate CLDR release 39 alpha0 to ICU trunk

srl295 commented:
@pedberg-icu  following the instructions in icu/tools/cldr/cldr-to-icu/README.txt I get this error:

```
$ ant -f build-icu-data.xml 
Buildfile: /Users/srl295/src/icu/tools/cldr/cldr-to-icu/build-icu-data.xml

init-args:

prepare-jar:
     [exec] [INFO] Scanning for projects...
     [exec] [INFO] 
     [exec] [INFO] --------------------< org.unicode.icu:cldr-to-icu >---------------------
     [exec] [INFO] Building cldr-to-icu 1.0-SNAPSHOT
     [exec] [INFO] --------------------------------[ jar ]---------------------------------
     [exec] [INFO] 
     [exec] [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ cldr-to-icu ---
     [exec] [INFO] Using 'UTF-8' encoding to copy filtered resources.
     [exec] [INFO] Copying 4 resources
     [exec] [INFO] 
     [exec] [INFO] --- maven-compiler-plugin:3.5.1:compile (default-compile) @ cldr-to-icu ---
     [exec] [INFO] Nothing to compile - all classes are up to date
     [exec] [INFO] 
     [exec] [INFO] --- maven-assembly-plugin:3.1.1:single (default) @ cldr-to-icu ---
     [exec] [INFO] Building jar: /Users/srl295/src/icu/tools/cldr/cldr-to-icu/target/cldr-to-icu-1.0-SNAPSHOT-jar-with-dependencies.jar
     [exec] [INFO] ------------------------------------------------------------------------
     [exec] [INFO] BUILD SUCCESS
     [exec] [INFO] ------------------------------------------------------------------------
     [exec] [INFO] Total time:  6.565 s
     [exec] [INFO] Finished at: 2021-02-02T10:16:16-06:00
     [exec] [INFO] ------------------------------------------------------------------------

clean:
[outputDirectories] Deleting 3,934 auto-generated files...

convert:

     [exec] Buildfile: /Users/srl295/src/icu/tools/cldr/cldr-to-icu/build-icu-data.xml
     [exec] 
     [exec] convert-impl:
     [exec]   [convert] Configuration lists redundant locale IDs
     [exec]   [convert] The following IDs should be removed from the configuration:
     [exec]   [convert] nb_NO, nb_SJ
     [exec]   [convert] Canonical ID list is:
     [exec]   [convert]     // A
     [exec] BUILD FAILED
     [exec] /Users/srl295/src/icu/tools/cldr/cldr-to-icu/build-icu-data.xml:142: java.lang.IllegalStateException: Non-canonical configuration
     [exec]     at org.unicode.icu.tool.cldrtoicu.ant.LocaleIdResolver.resolve(LocaleIdResolver.java:78)
     [exec]     at org.unicode.icu.tool.cldrtoicu.ant.LocaleIdResolver.expandTargetIds(LocaleIdResolver.java:30)
     [exec] 
     [exec]     at org.unicode.icu.tool.cldrtoicu.ant.ConvertIcuDataTask.execute(ConvertIcuDataTask.java:326)  [convert]     af, agq, ak, am, ar, ars, as, asa, ast, az, az_AZ, az_Cyrl
     [exec]     at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
     [exec]     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec] 
     [exec]   [convert]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
     [exec]   [convert]     // B
     [exec] 
     [exec]   [convert]     bas, be, bem, bez, bg, bm, bn, bo, br, brx, bs, bs_BA, bs_Cyrl      at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]   [convert] 
     [exec]     at java.base/java.lang.reflect.Method.invoke(Method.java:564)
     [exec]     at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
     [exec]     at org.apache.tools.ant.Task.perform(Task.java:350)
     [exec]     at org.apache.tools.ant.Target.execute(Target.java:449)
     [exec]     at org.apache.tools.ant.Target.performTasks(Target.java:470)
     [exec]     at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1401)
     [exec]     at org.apache.tools.ant.Project.executeTarget(Project.java:1374)
     [exec] 
     [exec]   [convert]     // C
     [exec]   [convert]     ca, ccp, ce, ceb, cgg, chr, ckb, cs, cy
     [exec]   [convert] 
     [exec]   [convert]     // D
     [exec]   [convert]     da, dav, de, dje, doi, dsb, dua, dyo, dz
     [exec]   [convert] 
     [exec]   [convert]     // E
     [exec]   [convert]     ebu, ee, el, en, en_NH, en_RH, eo, es, et, eu, ewo
     [exec]   [convert] 
     [exec]   [convert]     // F
     [exec]   [convert]     fa, ff, ff_Adlm, ff_CM, ff_GN, ff_MR, ff_SN, fi, fil, fo, fr, fur, fy
     [exec]   [convert] 
     [exec]   [convert]     // G
     [exec]   [convert]     ga, gd, gl, gsw, gu, guz, gv
     [exec]   [convert] 
     [exec]     at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
     [exec]     at org.apache.tools.ant.Project.executeTargets(Project.java:1264)
     [exec]   [convert]     // H
     [exec]   [convert]     ha, haw, he, hi, hr, hsb, hu, hy
     [exec]   [convert] 
     [exec]   [convert]     // I
     [exec]   [convert]     ia, id, ig, ii, in, in_ID, is, it, iw, iw_IL
     [exec]   [convert] 
     [exec]   [convert]     // J
     [exec]     at org.apache.tools.ant.Main.runBuild(Main.java:827)
     [exec]     at org.apache.tools.ant.Main.startAnt(Main.java:223)
     [exec]   [convert]     ja, jgo, jmc, jv    at org.apache.tools.ant.launch.Launcher.run(Launcher.java:284)
     [exec]   [convert] 
     [exec] 
     [exec]     at org.apache.tools.ant.launch.Launcher.main(Launcher.java:101)  [convert]     // K
     [exec]   [convert]     ka, kab, kam, kde, kea, khq, ki, kk, kkj, kl, kln, km, kn, ko, kok, ks
     [exec] 
     [exec] 
     [exec] Total time: 3 seconds
     [exec]   [convert]     ks_IN, ksb, ksf, ksh, ku, kw, ky
     [exec]   [convert] 
     [exec]   [convert]     // L
     [exec]   [convert]     lag, lb, lg, lkt, ln, lo, lrc, lt, lu, luo, luy, lv
     [exec]   [convert] 
     [exec]   [convert]     // M
     [exec]   [convert]     mai, mas, mer, mfe, mg, mgh, mgo, mi, mk, ml, mn, mni, mni_IN, mo, mr, ms
     [exec]   [convert]     mt, mua, my, mzn
     [exec]   [convert] 
     [exec]   [convert]     // N
     [exec]   [convert]     naq, nb, nd, ne, nl, nmg, nn, nnh, no, no_NO_NY, nus, nyn
     [exec]   [convert] 
     [exec]   [convert]     // O
     [exec]   [convert]     om, or, os
     [exec]   [convert] 
     [exec]   [convert]     // P
     [exec]   [convert]     pa, pa_Arab, pa_IN, pa_PK, pcm, pl, ps, pt
     [exec]   [convert] 
     [exec]   [convert]     // Q
     [exec]   [convert]     qu
     [exec]   [convert] 
     [exec]   [convert]     // R
     [exec]   [convert]     rm, rn, ro, rof, ru, rw, rwk
     [exec]   [convert] 
     [exec]   [convert]     // S
     [exec]   [convert]     sa, sah, saq, sat, sat_IN, sbp, sd, sd_Deva, sd_PK, se, seh, ses, sg, sh, sh_BA, sh_CS
     [exec]   [convert]     sh_YU, shi, shi_Latn, shi_MA, si, sk, sl, smn, sn, so, sq, sr, sr_BA, sr_CS, sr_Cyrl_CS, sr_Cyrl_YU
     [exec]   [convert]     sr_Latn, sr_Latn_CS, sr_Latn_YU, sr_ME, sr_RS, sr_XK, sr_YU, su, su_ID, sv, sw
     [exec]   [convert] 
     [exec]   [convert]     // T
     [exec]   [convert]     ta, te, teo, tg, th, ti, tk, tl, tl_PH, to, tr, tt, twq, tzm
     [exec]   [convert] 
     [exec]   [convert]     // U
     [exec]   [convert]     ug, uk, ur, uz, uz_AF, uz_Arab, uz_Cyrl, uz_UZ
     [exec]   [convert] 
     [exec]   [convert]     // V
     [exec]   [convert]     vai, vai_LR, vai_Latn, vi, vun
     [exec]   [convert] 
     [exec]   [convert]     // W
     [exec]   [convert]     wae, wo
     [exec]   [convert] 
     [exec]   [convert]     // X
     [exec]   [convert]     xh, xog
     [exec]   [convert] 
     [exec]   [convert]     // Y
     [exec]   [convert]     yav, yi, yo, yue, yue_CN, yue_HK, yue_Hans
     [exec]   [convert] 
     [exec]   [convert]     // Z
     [exec]   [convert]     zgh, zh, zh_CN, zh_HK, zh_Hant, zh_MO, zh_SG, zh_TW, zu
     [exec]   [convert] 

BUILD FAILED
/Users/srl295/src/icu/tools/cldr/cldr-to-icu/build-icu-data.xml:109: exec returned: 1

Total time: 12 seconds
```

srl295 commented:
I got further with this change, but i haven't put it into the PR:

```diff
diff --git a/tools/cldr/cldr-to-icu/build-icu-data.xml b/tools/cldr/cldr-to-icu/build-icu-data.xml
index ea2f6e3f87..f1b9af507b 100644
--- a/tools/cldr/cldr-to-icu/build-icu-data.xml
+++ b/tools/cldr/cldr-to-icu/build-icu-data.xml
@@ -197,7 +197,7 @@
                 mt, mua, my, mzn
 
                 // N
-                naq, nb, nb_NO, nb_SJ, nd, ne, nl, nmg, nn, nnh, no, no_NO_NY, nus, nyn
+                naq, nb, nd, ne, nl, nmg, nn, nnh, no, no_NO_NY, nus, nyn
 
                 // O
                 om, or, os
@@ -290,7 +290,7 @@
 
                     // K-P
                     ka, kk, kl, km, kn, kok, ko, ku, ky, lb, lkt, ln, lo, lt, lv,
-                    mk, ml, mn, mo, mr, ms, mt, my, nb, nb_NO, ne, nl, nn, no, no_NO,
+                    mk, ml, mn, mo, mr, ms, mt, my, nb, ne, nl, nn, no, no_NO,
                     om, or, pa_IN, pa, pa_Guru, pl, ps, pt,
 
                     // R-T
```

srl295 commented:
@hagbard it seems icu4c/source/data/build.xml is *completely* obsolete, and I delete it here.  It seems the "generate production data" part that was all that was left is not needed because cldr-to-icu already consumes production data from cldr-staging.

hagbard commented:
> @hagbard it seems icu4c/source/data/build.xml is _completely_ obsolete, and I delete it here. It seems the "generate production data" part that was all that was left is not needed because cldr-to-icu already consumes production data from cldr-staging.

No issues from me! (sorry for not being more attentive to this recently, I just started ramping up on a new project that's taking all my time).

pedberg-icu commented:
Thanks for this. I have the integration working with the current process in my main PR (most of the work not pushed yet). Let me fix some test failures on that, then I will take a closer look at this.

sffc committed:
ICU-21067 Maybe fix synchronization lint in MeasureUnit.java

sffc committed:
ICU-21484 Add SignDisplay NEGATIVE

echeran committed:
ICU-21310 Consolidate ICU4C and ICU4J Readmes via the User Guide

erik0686 committed:
ICU-21480 BRS69RC Version update and regenerate configure for v69

See #1567

sffc committed:
ICU-20421 Add span fields to NumberRangeFormatter

See #1572

richgillam committed:
ICU-21480 Promoted all @draft ICU 67 APIs to @stable ICU 67.

aheninger committed:
ICU-21492 Fix regex compile assertion failure.

A regex pattern containing nested look-behind blocks could trigger an assertion
failure during pattern compilation. The problem was caused by an off-by-one
error in the code that computes an upper bound on the match length, needed
because look-behind expressions are constrained to not have unbounded match
length.

Nested look-behind blocks come into play because, when computing the maximum
match length of an outer block, any inner look-behind blocks are skipped over -
they do not directly contribute to the length matched by the outer block. The
problem was in the code that skips over these nested look-behind blocks.

pedberg-icu committed:
ICU-21480 integrate CLDR release 39 alpha1 to ICU trunk

hugovdm committed:
ICU-21284 Deal with remaining unit normalization TODOs

See #1580

markusicu committed:
ICU-13702 add missing API functions
and fix a bug in Java UnicodeSet.retain(String) which added the string even if the set did not contain it before,
and some drive-by API doc fixes/clarifications

hugovdm committed:
ICU-20941 NumberFormatter: format arbitrary compound units, with inflections

See #1588.

hugovdm committed:
ICU-21123 Support unit inflections in ICU4C

See #1574

jefgen committed:
ICU-21499 Suppress warnings about non-NULL address in uprv_memcpy and uprv_memmove

Commit:
ICU-21487 Moves performance test data files from the icu-data repository
to the icu repository. Added Unicode copyright notice to those files
that only have the IBM copyright notice.
A complimentary PR will remove data from the icu-data repository.

ICU-21487 Adds pure test data files which have no copyright notice to the
exemption list.

sffc committed:
ICU-21203 ICU-21198 ICU-21479 Improve docs discoverability for NumberFormatter

See #1559

younies committed:
ICU-21349 Add extra ComplexUnitsConverter constructor that takes only CLDR units Identifier

See #1586

pedberg-icu committed:
ICU-21480 integrate CLDR release-39-alpha2 to ICU trunk

Commit:
ICU-21323 Automates the uconfig variation BRS tasks. The test
subsequently sets each of the UCONFIG_NO_XXX flags to '1' (exceptions apply)
and runs the ICU4C unit tests and the header test. Afterwards all
UCONFIG_NO_XXX flags are set to '1' and unit tests and header test
are executed.

To allow concurrent execution and reduce total run time, the script provides
the option to execute only the unit tests or only the header test.

ICU-21323 Split the uconfig.h variation test into two to reduce
execution time. One test now tests the variations with unit tests, the
other tests with header test.
Execution time now ranges between 45 and 58 minutes.

ICU-21323 Works in review comments.

ICU-21323 Factors in more review comments.

srl295 committed:
ICU-20233 tool: improvements to commit checker for CLDR

- improve checks for CLDR resolutions
- update batch fetch from 50 to 100
- do not show no-commit tickets under illegal-state
- add navigation
- use numeric IDs for status, resolution, types with constants
 (S_, R_, I_)

fix bugs from CLDR-14224
- do not show not-in-query issues as open
- cache single-issue queries
- give information on each problem as to what it is
- put 'review' in a separate section (not counted in Problems)
- more detailed fix info under each issue

- fix section headings, always show component, add nocopyright option
- sort open issues by componnent
- sort the 'review' list

ICU-20444:
- exclude (via git-cherry) any commits already cherry picked to the branch.
- used a workaround to https://github.com/gitpython-developers/GitPython/issues/846
that seemed to work well,
- warn about issues with excluded commits

CLDR-14524:
- Exclude "Obsolete" from requiring commits

Commit:
ICU-21502 Adds status checks to test to prevent segementation fault when
test runs with ICU stub data only.

ICU-21502 Review comments worked in.

hugovdm committed:
ICU-21432 Fix TestInternalMeasureUnitImpl() segfaulting with bad data

younies committed:
ICU-21349 Add extra UnitsRouter constructor that takes only CLDR unit…

See #1579

Commit:
ICU-21503 Adds status checks to prevent segmentation fault when test runs
with stubdata only.

ICU-21503 Changes the no-data checks according to review feedback.

hugovdm committed:
ICU-20941 Fix ResourceTable lifetime to make ResourceTracer happy

This is a separate commit from the previous ICU-20941 commit for the
sake of documentation / future code archaeology.

See #1588

younies committed:
ICU-21349 Add extra UnitsConverter constructor that takes only CLDR unit identifiers

See #1578

pedberg-icu committed:
ICU-21480 integrate CLDR release-39-alpha3 to ICU trunk

abarberenaCPDS committed:
ICU-21416 Update icu4c-readme.md

Syntax error, it should be:

git clone https://github.com/unicode-org/icu.git

pxeger committed:
ICU-21416 Fix markdown formatting in userguide/strings

younies committed:
ICU-21123 Support unit inflections in ICU4J

See #1590

hugovdm committed:
ICU-20941 Fix uninitialized values: DerivedComponents' compound0_ and compound1_

jefgen committed:
ICU-21480 Update GitHub pull request template: JIRA issue number is needed on commit messages.

hugovdm committed:
ICU-21480 Update French unit-times pattern in unit tests

sffc committed:
ICU-20019 Implement withSignificantDigits option

See #1598

sffc committed:
ICU-21493 Add more rounding modes in ICU4C

See #1575

pedberg-icu committed:
ICU-21480 integrate CLDR release-39-alpha4 to ICU trunk

sven-oly committed:
ICU-21179 Remove unused code in Java break engines

See #1604

yumaoka committed:
ICU-21370 Delete folder for eclipse plug-in build

Commit:
ICU-21323 Adds BRS task 'test ICU4C without data' to GHA, triggered by
each pull request. This test has a high catch rate.

ICU-21323 Work in review feedback.

jefgen committed:
ICU-21480 Update GitHub pull request template to prevent autolinking example ticket number.

mojca committed:
ICU-21513 check if TARGET_OS_SIMULATOR has been defined

clang 11 throws an error when using undefined variables starting with
TARGET_OS_, and TARGET_OS_SIMULATOR is missing in earlier versions
of macOS, so the code would not compile without an additional guard.

See also ICU-13440.

sffc committed:
ICU-20886 Implement trailingZeroDisplay

See #1583

hugovdm committed:
ICU-20941 Port Arbitrary Units support from ICU4C to ICU4J

See #1597

hugovdm committed:
ICU-21349 ICU4J units cleanup: eclipse warnings, unsuppress one warning

toString() now uses category, so @SuppressWarnings("unused") is no
longer needed.

yumaoka committed:
ICU-21490 Exposing getOffsetFromLocal for ICU4J

See #1609

yumaoka committed:
ICU-21372 getOffsetFromLocal for C and C++

See #1610

Commit:
ICU-21323 Automates BRS testing tasks of U_CHARSET_IS_UTF8 and
U_OVERRIDE_CXX_ALLOCATION=0.

markusicu committed:
ICU-21368 unit-test & fix BytesTrie jump delta encoding

FrankYFTang committed:
ICU-21222 Fix Era in Interval format

See #1373

erik0686 committed:
ICU-21480 BRS69RC update readmes for RC

See #1615

hugovdm committed:
ICU-21123 Support FormattedNumber::getGender() for "short" and "narrow" formatting too

See #1617

yumaoka committed:
ICU-21369 Remove @provisional API doc tags in ICU4J

echeran committed:
ICU-21519 Add PluralOperand 'c' as alias to 'e', parse FixedDecimal strings

Commit:
ICU-21323 Automates BRS test task of build and run testmap.

ICU-21323 Removes compiler flag.

FrankYFTang committed:
ICU-21521 Fix cast of uprv_strcmp

See #1618

younies committed:
ICU-21349 calling .usage("") should unset the existing usage

See #1614

FrankYFTang committed:
ICU-21127 Error when rbbi got unpaired surrogate char

See #1520

hugovdm committed:
ICU-21349 Delete MeasureUnitImpl copy constructor, drop an indirection

hugovdm committed:
ICU-21313 UserGuide: use absolute GitHub links for links outside docs/

See #1625

pedberg-icu committed:
ICU-21480 brs 69rc, update urename.h pass 0

yumaoka committed:
ICU-21507 Removing unused class initializer code

ICUDebug seemed to check Java version whether it's 1.4 or higher. The code referencing the version is already gone. We also no longer support Java 1.3 runtime since long time ago. Remove the unused code checking Java version.

yumaoka committed:
ICU-21498 Improve comment in uprv_tzname

yumaoka committed:
ICU-21207 Remove unnecessary reflection call

The test code was written long time ago. Java 1.3 TimeZone did not have getDSTSavings() method. The test code compares DST saving of Lord_Howe between ICU and JDK, and calling getDSTSavings() through reflection for 1.3 compatibility. This is no longer necessary and replaced with regular method call.

sffc committed:
ICU-21397 Allow null to unset options in Java NumberRangeFormatter

yumaoka committed:
ICU-20918 ucal_openTimeZones() to call ucal_openTimeZoneIDEnumeration()

See #1630

sven-oly committed:
ICU-21480 BRS#18 ICU4J promotions to stable

hugovdm committed:
ICU-21123 Calculate built-in units' gender when CLDR lacks the data

See #1620

pedberg-icu committed:
ICU-21480 brs 69rc, add tests to verify that rbnf spellout is the same for no,nb

sffc committed:
ICU-21330 Use =0 and =1 plural forms in compact notation

See #1632

FrankYFTang committed:
ICU-21041 Fix fuzzer memory read error.

See #1622

sven-oly committed:
ICU-21480 Change ICU 64 draft readResolve APIs to @internal

See #1637

richgillam committed:
ICU-21335 Disallowed looking up resources by index in table-based resources.

FrankYFTang committed:
ICU-21534 Add notice about master to main rename

See #1638

pedberg-icu committed:
ICU-21341 for getISO codes, add some CLDR-valid regions, remove some CLDR-invalid langs

pedberg-icu committed:
ICU-21532 In UTimeZoneLocalOption enum, add dummy value if U_HIDE_DRAFT_API

pedberg-icu committed:
ICU-21480 integrate CLDR release-39-beta to ICU trunk

sven-oly committed:
ICU-21480 Update ICU4C APIChangeReport files

sven-oly committed:
ICU-21480 Update ICU4J APIChangeReport.html to 69

pedberg-icu committed:
ICU-21480 brs 69rc, adjust nonstable API macros

yumaoka committed:
ICU-21480 BRS69 clean up import statements

yumaoka committed:
ICU-21480 BRS69 ICU4J 69 serialization test data

Also dropped ICU4J 64 test data.

yumaoka committed:
ICU-21480 BRS69 ICU4J API signature file

- Supplied missing API comment to com.ibm.icu.nubmer.Precision#trailingZeroDisplay(TrailingZeroDisplay)
- Generated API signature file for ICU4J 69.1 and added to the repository
- Delete ICU4J 59 API signature file
- Deleted unused icu4j/eclipse-build/build.properties. The folder eclipse-build was deleted, but someone worked on ICU version seemed to add it back again. Delete it again.

sven-oly committed:
ICU-21480 BRS-50: updated docmain.h

See #1647

sven-oly committed:
ICU-21480 Scrub TODOs for completed issues 21292 and 21236

See #1652

echeran committed:
ICU-21480 fix logKnownIssue number

echeran committed:
ICU-21480 Change ICU4C/J Readme update date after version change

Commit:
ICU-21520 Fixes typo in name of test data file; removes a regex that
evaluates to the empty string and the result files are missing the
perf test name as a result. E.g. '_Mar_16_155820.html' instead of
'unisetperf_Mar_16_155820.html'.
Not obvious why the regex was even created since it doesn't seem
necessary because the name of the caller doesn't require additional
parsing. The rpaticular code is from 2002, maybe changes in Perl
account for the issue(?)

FrankYFTang committed:
ICU-21537 Fix invalid free by long locale name

Do not free baseName if it is pointing to fullNameBuffer.

Better Fix

sffc committed:
ICU-21480 Update double-conversion

jefgen committed:
ICU-21542 Have GHA output ICU4J test failures if there are any.

richgillam committed:
ICU-21552 Changed to only display "Encountered empty table" warning in verbose mode.

hugovdm committed:
ICU-21508 Rename 'master' to 'main' in docs, maven files, scripts, etc

See #1664

longnguyen2004 committed:
ICU-21560 mingw: Remove version numbers from link flags

FrankYFTang committed:
ICU-21550 Add zzzz to subdivision if len==2

See #1662

FrankYFTang committed:
ICU-21526 Change return of UElementComparator to int32_t

See #1621

srl295 committed:
ICU-21568 update pom.xml to point to CLDR maven

- updated README as well
- deleted install-cldr-jars.sh
